test/SlowTests/Issues/RavenDB-18554.cs

```diff
@@ -185,15 +185,31 @@ public async Task QueriesShouldFailoverIfIndexIsCompacting(bool cluster = false)
                 {
                     await session.StoreAsync(c);
                     if (cluster)
-                        session.Advanced.WaitForReplicationAfterSaveChanges(replicas: 1);
+                        session.Advanced.WaitForReplicationAfterSaveChanges(replicas: 1); //Wait for replication
                     await session.SaveChangesAsync();
                     categoryId = c.Id;
                 }
 
+                // Wait for indexing in first node
                 var index = new Categoroies_Details();
                 index.Execute(store);
                 Indexes.WaitForIndexing(store);
 
+                // Wait for indexing in second node (if it's cluster test)
+                if (cluster)
```

Comment:
I do not like this approach. Create a separate test for Cluster and separate one for single node. You even have different asserts for those. All stores that need to perform an action on Exact node in cluster should have topology updates disabled. The store that does the query should have them enabled. Now this test does not have control over what is going on, so I'm assuming that the conditions are changing, one run it will do work on node A, second one on node B, etc.

Reply:
Ok

Reply:
Done